### DISCUSSION TITLE
Backslashes in links and images

### DISCUSSION THREAD

**Discussion by kangruixiang** [remarkjs/remark#549] [Q&A] [answered]
Not to overwhelm you with issues. Here's something else I found: 

Using the new micromark, backslashes are added within names/links, example: https://runkit.com/kangruixiang/5f908ad22c9682001a7ccff6

This kind of breaks the link/image link. Not sure if this was intended?

**Comment by wooorm** [comment, marked as answer]
> Not to overwhelm you with issues.

💚

> Not sure if this was intended?

This is intended, so that emphasis cannot accidentally form.

> This kind of breaks the link/image link.

Does it really? The escapes work in text (`6\_1` and `6_1` are equivalent between `[]`) and in the destination (the url, where `6\_1` and `6_1` are also equivalent).

**Reply by kangruixiang** [reply]
So the links still work in Obsidian but when opened up in Typora, the image link is broken, at least from my testing. Right now I'm just adding regex to remove the extra backslash after conversion each time.

**Reply by wooorm** [reply]
hmm, that is unfortunate: that Typora is not CommonMark compliant. You may want to point Typora folks to remark/micromark/another commonmark compliant parser :wink:

**Reply by kangruixiang** [reply]
well thanks for the help still.

**Reply by bennycode** [reply]
@wooorm I’ve noticed that links containing escaped underscores in Markdown files don’t work when clicked in VS Code.

**Reply by ChristianMurphy** [reply]
@bennycode sounds like a bug in vscode

**Reply by remcohaszing** [reply]
VSCode provides autolink detection for any text files. It doesn’t interpret the context where the link appears. I wouldn’t say this is a bug, neither in micromark nor in VSCode, but it is inconvenient.

**Reply by bennycode** [reply]
Thanks for getting back to me. 🙂 I am new to "remark" and I just installed it so I can use a custom plugin (which uses 'unist-util-visit' to parse and modify 'code' blocks). I was surprised that a lot of other changes were then made by 'remark'. Is there any chance to deactivate all the other changes (like bullet list style changes, link escaping, etc.)?

**Reply by wooorm** [reply]
Take some time with the docs: they talk about ASTs and what that means. Changes are intentional. You can configure which style to use for lists. Escaping things is needed, just like how a `"` has to be escaped in a JavaScript `"string"`.

If you post your actual content, more info, we can better help you with your root problem. Perhaps you can manually format your markdown. Perhaps there is a bug somewhere. Perhaps we can explain why things happen!
